include/mlir/Dialect/SPIRV/SPIRVCastOps.td

```diff
@@ -0,0 +1,281 @@
+//===-- SPIRVCastOps.td - MLIR SPIR-V Cast Ops -*- tablegen -*-===//
+//
+// Copyright 2019 The MLIR Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =============================================================================
+//
+// This file contains cast ops for the SPIR-V dialect. It corresponds
+// to "3.32.11. Convertion Instructions" of the SPIR-V specification.
+//
+//===----------------------------------------------------------------------===//
+
+#ifdef SPIRV_CAST_OPS
+#else
+#define SPIRV_CAST_OPS
+
+#ifdef SPIRV_BASE
+#else
+include "mlir/SPIRV/SPIRVBase.td"
+#endif // SPIRV_BASE
+
+class SPV_CastOp<string mnemonic, Type resultType, Type operandType,
+                 list<OpTrait> traits = []> :
+      SPV_Op<mnemonic,
+             !listconcat(traits,
+                         [NoSideEffect, SameOperandsAndResultShape])> {
+  let arguments = (ins
+    SPV_ScalarOrVectorOf<operandType>:$operand
+  );
+
+  let results = (outs
+    SPV_ScalarOrVectorOf<resultType>:$result
+  );
+
+  let parser = [{ return mlir::impl::parseCastOp(parser, result); }];
```

Comment:
This is inconsistent with spv.Bitcast. Can you upate spv.Bitcast to also use this?